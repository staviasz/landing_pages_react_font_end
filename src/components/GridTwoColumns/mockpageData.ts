export const data = {
  sectionId: 'id',
  title: 'Title',
  text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Consequuntur ad consequatur numquam possimus quaerat reprehenderit dolore, doloribus corrupti delectus accusamus maxime id maiores iste eveniet praesentium, dolor nisi, sapiente nemo.',
  srcImage: 'https://cdn-icons-png.flaticon.com/512/87/87390.png',
  background: false,
};
