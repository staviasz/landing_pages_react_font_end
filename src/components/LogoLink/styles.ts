import styled, { css } from 'styled-components';

export const LinkContent = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
    > img {
      max-height: 3rem;
    }
  `}
`;
