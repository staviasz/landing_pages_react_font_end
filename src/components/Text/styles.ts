import styled, { css } from 'styled-components';

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    text-align: left;
  `}
`;
