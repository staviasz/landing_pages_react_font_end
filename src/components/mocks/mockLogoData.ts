export const logoData = {
  text: 'Logo',
  link: '#target',
  srcImage: 'https://cdn-icons-png.flaticon.com/512/87/87390.png',
};
