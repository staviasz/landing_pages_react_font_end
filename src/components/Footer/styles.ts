import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Text } from '../Text/styles';

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding: ${theme.spacings.xsmall};
      @media ${theme.media.lteMedium} {
        max-width: 80%;
        margin-left: 18px;
      }
    }

    ${Text} {
      font-size: ${theme.font.sizes.small};
      text-align: center;
      margin: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
