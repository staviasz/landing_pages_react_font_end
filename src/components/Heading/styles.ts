import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface TitleProps {
  colordark: string;
  size: string;
  uppercase: string;
}

interface TitleSize {
  [key: string]: (theme: Theme) => ReturnType<typeof css>;
}

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleSize: TitleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xhuge: (theme) => css`
    ${mediaFont(theme)}
    font-size: ${theme.font.sizes.xhuge};
  `,
};

export const Title = styled.h1<TitleProps>`
  ${({ theme, colordark, size, uppercase }) => css`
    color: ${colordark === 'dark'
      ? theme.colors.primaryColor
      : theme.colors.white};
    ${titleSize[size](theme)}
    text-transform: ${uppercase};
    font-family: ${theme.font.family.secondary};
    font-style: italic;
  `}
`;
