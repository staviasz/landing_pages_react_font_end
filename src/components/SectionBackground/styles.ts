import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface ContainerProps {
  background: string;
}

const containerBackgroundActivate = (theme: Theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div<ContainerProps>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background === 'true' && containerBackgroundActivate(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
