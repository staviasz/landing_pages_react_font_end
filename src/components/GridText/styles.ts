import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }
    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
