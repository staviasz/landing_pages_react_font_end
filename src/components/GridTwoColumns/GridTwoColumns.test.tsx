import { GridTwoColumn } from '.';
import { renderTheme } from '../../styles/render-theme';
import { data } from '../mocks/mockpageData';

describe('<GridTwoColumn />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumn {...data} />);
    expect(container).toMatchSnapshot();
  });
});
