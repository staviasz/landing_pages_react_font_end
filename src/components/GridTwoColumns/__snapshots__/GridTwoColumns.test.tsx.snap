// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
  font-family: 'Montserrat',sans-serif;
  font-style: italic;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c3 {
  min-width: 280px;
}

.c7 {
  text-align: center;
}

.c8 {
  max-width: 500px;
  width: 100%;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Title
          </h1>
          <p
            class="c6"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Consequuntur ad consequatur numquam possimus quaerat reprehenderit dolore, doloribus corrupti delectus accusamus maxime id maiores iste eveniet praesentium, dolor nisi, sapiente nemo.
          </p>
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title"
            class="c8"
            src="https://cdn-icons-png.flaticon.com/512/87/87390.png"
          />
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
